hedera-node/hedera-mono-service/src/main/java/com/hedera/services/txns/crypto/CryptoCreateTransitionLogic.java

```diff
@@ -132,17 +163,100 @@ private HederaAccountCustomizer asCustomizer(CryptoCreateTransactionBody op) {
         long consensusTime = txnCtx.consensusTime().getEpochSecond();
         long expiry = consensusTime + autoRenewPeriod;
 
-        /* Note that {@code this.validate(TransactionBody)} will have rejected any txn with an invalid key. */
-        final JKey key = asFcKeyUnchecked(op.getKey());
-        HederaAccountCustomizer customizer =
-                new HederaAccountCustomizer()
-                        .key(key)
-                        .memo(op.getMemo())
-                        .expiry(expiry)
-                        .autoRenewPeriod(autoRenewPeriod)
-                        .isReceiverSigRequired(op.getReceiverSigRequired())
-                        .maxAutomaticAssociations(op.getMaxAutomaticTokenAssociations())
-                        .isDeclinedReward(op.getDeclineReward());
+        var customizer = new HederaAccountCustomizer();
+        if (!op.getAlias().isEmpty() && !op.hasKey()) {
+            final var resolution = aliasManager.lookupIdBy(op.getAlias());
+            if (!resolution.equals(MISSING_NUM)) {
+                throw new InvalidTransactionException(INVALID_ALIAS_KEY);
+            }
+
+            if (op.getAlias().size() == EVM_ADDRESS_SIZE) {
+                customizer
+                        .memo(AUTO_MEMO)
+                        .autoRenewPeriod(THREE_MONTHS_IN_SECONDS)
+                        .expiry(txnCtx.consensusTime().getEpochSecond() + THREE_MONTHS_IN_SECONDS)
+                        .isReceiverSigRequired(false)
+                        .isSmartContract(false)
+                        .alias(op.getAlias());
+            } else {
+                final var keyFromAlias = asPrimitiveKeyUnchecked(op.getAlias());
+                if (!keyFromAlias.getECDSASecp256K1().isEmpty()) {
+                    final var recoveredEvmAddress =
```

Comment:
Move the `recoveredEvmAddress` calculation and check into one method as the logic is duped at least 2 more times